src/app/components/feed/FeedActions.js

```diff
@@ -0,0 +1,260 @@
+import React, { useState } from 'react';
+import { createFragmentContainer, graphql } from 'react-relay/compat';
+import PropTypes from 'prop-types';
+import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import styles from './SaveFeed.module.css';
+import Can from '../Can';
+import Alert from '../cds/alerts-and-prompts/Alert';
+import ButtonMain from '../cds/buttons-checkboxes-chips/ButtonMain';
+import ChevronDownIcon from '../../icons/chevron_down.svg';
+import ConfirmProceedDialog from '../layout/ConfirmProceedDialog';
+
+const FeedActions = ({
+  saving,
+  handleDelete,
+  handleLeaveFeed,
+  feedTeam,
+}) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [showDeleteDialog, setShowDeleteDialog] = React.useState(false);
+  const [showLeaveDialog, setShowLeaveDialog] = useState(false);
+  const { feed } = feedTeam;
+  const isFeedOwner = feedTeam.team_id === feed.team.dbid;
+
+  const handleMenuClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleDeleteClick = () => {
+    setShowDeleteDialog(true);
+  };
+
+  const handleLeaveClick = () => {
+    setShowLeaveDialog(true);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+    setShowDeleteDialog(false);
+    setShowLeaveDialog(false);
+  };
+
+  let menuItem = null;
+
+  if (!isFeedOwner) {
+    menuItem = (
+      <MenuItem onClick={handleLeaveClick}>
+        <FormattedMessage
+          id="SaveFeed.leaveButton"
+          defaultMessage="Leave shared feed"
+          description="Menu option to leave the selected shared feed"
+        />
+      </MenuItem>
+    );
+  } else if (feed.teams_count > 1) {
+    menuItem = (
+      <MenuItem disabled>
+        <FormattedHTMLMessage
+          id="SaveFeed.deleteButtonDisabled"
+          defaultMessage="Delete shared feed <br />(Remove collaborators to <br /> delete this shared feed)"
+          description="Menu option to inform user to remove collaborators before deleting the selected shared feed"
+        />
+      </MenuItem>
+    );
+  } else {
+    menuItem = (
+      <MenuItem onClick={handleDeleteClick}>
+        <FormattedMessage
+          id="SaveFeed.deleteButton"
+          defaultMessage="Delete shared feed"
+          description="Menu option to delete the selected shared feed"
+        />
+      </MenuItem>
+    );
+  }
+
+  return (
+    <>
+      {/* FIXME: Review actual permissions for creator org vs invited org */}
+      <Can permissions={feed.permissions} permission="destroy Feed">
```

Comment:
@caiosba hopefully `destroy Feed` really is the right permission here?

Reply:
Yes for deleting a feed. For an invited organization to leave a feed, then it's a `destroy FeedTeam` permission.